### PR TITLE
SISRP-13763 Back end for advising resources deep links

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.berkeley.edu/browse/SISRP-13763

Another day, another proxy.

### CODE REVIEW

File: fixtures/xml/campus_solutions/advising_resources.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        <NAME>Academic Requirements</NAME>
+        <URL>https://bcs-web-dev-03.is.berkeley.edu:8443/psc/bcsdev/EMPLOYEE/HRMS/c/MANAGE_ACADEMIC_RECORDS.STDNT_ACTIVATION.GBL?EMPLID=17154428</URL>
+        <IS_CS_LINK type="boolean">true</IS_CS_LINK>
+      </UC_ACADEMIC_REQUIREMENTS>
+      <UC_ADVISEE_STUDENT_CENTER>
+        <NAME>Advisee Student Center</NAME>
+        <URL>https://bcs-web-dev-03.is.berkeley.edu:8443/psc/bcsdev/EMPLOYEE/HRMS/c/SSR_ADVISEE_OVRD.SSS_STUDENT_CENTER.GBL?</URL>
+        <IS_CS_LINK type="boolean">true</IS_CS_LINK>
+      </UC_ADVISEE_STUDENT_CENTER>
+      <UC_ADVISING_FAVORITE_REPORTS type="array"/>
```
Comment (christianvuerings) [inline fixtures/xml/campus_solutions/advising_resources.xml]: @paulkerschen might be good if we could add someone in here who actually has favorite reports. Let me try to find someone.

Comment (christianvuerings) [inline fixtures/xml/campus_solutions/advising_resources.xml]: How about:
https://bcs-web-dev-03.is.berkeley.edu:8443/PSIGW/RESTListeningConnector/PSFT_CS/UC_AA_ADVISING_RESOURCES.v1/UC_ADVISING_RESOURCES?EMPLID=3030556855

And with the optional parameter:
https://bcs-web-dev-03.is.berkeley.edu:8443/PSIGW/RESTListeningConnector/PSFT_CS/UC_AA_ADVISING_RESOURCES.v1/UC_ADVISING_RESOURCES?EMPLID=3030556855&STUDENT_EMPLID=17154428

FYI, you can add your own favorites at
https://bcs-web-dev-03.is.berkeley.edu:8443/psc/bcsdev/EMPLOYEE/HRMS/s/WEBLIB_PTAL.ISCRIPT1.FieldFormula.IScript_PTAL_DASHBOARD?tab=UC_CS_RP_RC_RPT_DASHBOARD

**Comment by christianvuerings** [comment]
lgtm - just added a suggestion to add someone with favorite reports

**Comment by paulkerschen** [comment]
Thanks for the suggestion, @christianvuerings; I'll update the fixture.

**Comment by paulkerschen** [comment]
Amended to include favorites reports.

File: fixtures/xml/campus_solutions/advising_resources.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <NAME>Academic Requirements</NAME>
+        <URL>https://bcs-web-dev-03.is.berkeley.edu:8443/psc/bcsdev/EMPLOYEE/HRMS/c/MANAGE_ACADEMIC_RECORDS.STDNT_ACTIVATION.GBL?EMPLID=17154428</URL>
+        <IS_CS_LINK type="boolean">true</IS_CS_LINK>
+      </UC_ACADEMIC_REQUIREMENTS>
+      <UC_ADVISEE_STUDENT_CENTER>
+        <NAME>Advisee Student Center</NAME>
+        <URL>https://bcs-web-dev-03.is.berkeley.edu:8443/psc/bcsdev/EMPLOYEE/HRMS/c/SSR_ADVISEE_OVRD.SSS_STUDENT_CENTER.GBL?</URL>
+        <IS_CS_LINK type="boolean">true</IS_CS_LINK>
+      </UC_ADVISEE_STUDENT_CENTER>
+      <UC_ADVISING_FAVORITE_REPORTS type="array">
```
Comment (christianvuerings) [inline fixtures/xml/campus_solutions/advising_resources.xml]: @paulkerschen so sorry about this, but the structure doesn't seem to have been correct. I just modified it in CS and also informed Pradeep.

If you pull it again, everything should be fine now. You should see that `UC_ADVISING_FAVORITE_REPORTS` is on the same level as `UC_ADVISING_LINKS`

Comment (christianvuerings) [inline fixtures/xml/campus_solutions/advising_resources.xml]: image

**Comment by paulkerschen** [comment]
The back-end passthrough pattern is really paying off with these amendments! Thanks, @christianvuerings, fixture and dummy JSON changed one more time.

**Comment by christianvuerings** [comment]
@paulkerschen looks perfect, thank you so much for putting in the additional changes.

**Pull request merged**